.github/workflows/coverity.yml

```diff
@@ -0,0 +1,32 @@
+# Creates and uploads a Coverity build on a schedule
+name: Coverity Scan
+on:
+  #schedule:
+  #  # Run twice daily, since below 500k LOC can have 21 builds per week, per https://scan.coverity.com/faq#frequency
+  #  - cron: '0 */12 * * *'
+  # Support manual execution
+  workflow_dispatch:
+jobs:
+  coverity:
+    if: github.repository == 'redis/redis'
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@main
+    - name: Download and extract the Coverity Build Tool
+      run: |
+          wget -q https://scan.coverity.com/download/cxx/linux64 --post-data "token=${{ secrets.COVERITY_SCAN_TOKEN }}&project=redis" -O cov-analysis-linux64.tar.gz
+          mkdir cov-analysis-linux64
+          tar xzf cov-analysis-linux64.tar.gz --strip 1 -C cov-analysis-linux64
+    - name: Install Redis dependencies
+      run: sudo apt install -y gcc tcl8.6 tclx procps libssl-dev
+    - name: Build with cov-build
+      run: cov-analysis-linux64/bin/cov-build --dir cov-int make
+    - name: Upload the result
+      run: |
+          tar czvf cov-int.tgz cov-int
+          curl \
+            --form project=redis \
+            --form email=${{ secrets.COVERITY_SCAN_EMAIL }} \
+            --form token=${{ secrets.COVERITY_SCAN_TOKEN }} \
```

Comment:
so the reports will be received by email?
what does the other token do?

Reply:
@oranagra : My apologies, I should have been clearer.

> so the reports will be received by email?
> what does the other token do?

the `secrets.COVERITY_SCAN_EMAIL`  and `secrets.COVERITY_SCAN_TOKEN` are used as authentication parameters to upload the intermediate build files to Coverity, which would then trigger a Coverity scan. Reports are visible in the Coverity web portal (e.g., https://scan.coverity.com/projects/moshekaplan-redis?tab=overview ).

> i assume we should expect a clean report after enabling it, right?

Unfortunately not. Coverity detected about 290 issues; I had only submitted fixes for ones that seemed unambiguous and the fixes looked clear. If you'd like, you can see the completed CI job [here](https://github.com/moshekaplan/redis/actions/runs/6996510664/job/19032467044),  and the Coverity report for my fork here: https://scan.coverity.com/projects/moshekaplan-redis?tab=overview

Reply:
I remember some fixes for coverity can issues in the past were not real problems so they were rejected IIRC. But probably some of them are real issues.

I don't know who has registered "redis" in coverity. It's not one of us I think.

Reply:
I was incorrect, `secrets.COVERITY_SCAN_EMAIL` is not used as an authentication parameter, rather it is the notification email for the scan results. However, it is a required parameter - it cannot be skipped or left blank.